treeherder/perf/alerts.py

```diff
@@ -45,12 +46,12 @@ def generate_new_alerts_in_series(signature):
         series = series.filter(
             result_set_id__gt=existing_alerts[0].summary.result_set_id)
 
-    a = Analyzer()
+    a = []
     for datum in series:
         timestamp = int(time.mktime(
             datum.push_timestamp.timetuple()))
-        a.add_data(timestamp, datum.value,
-                   testrun_id=datum.result_set_id)
+        a.append(Datum(timestamp, datum.value,
+                       testrun_id=datum.result_set_id))
```

Comment:
I suspect you're getting an exception here because you need to sort the list of performance datums (which the analyzer class did internally). We should add some code in analyze to do that (see comment below).

'a' isn't a great name for the data, could you rename it to 'data'?

Also, now that I look at it, you could also make this a bit simpler by using a list comprehension:

```py
data = [Datum(int(time.mktime(d.push_timestamp.timetuple())), d.value, testrun_id=d.result_set_id) for d in series]
```